Improvements and bugfixes

petrparolek opened the pull request:
- fixed tests
- fixed code reported by PHPStan
- upgrade PHPStan

petrparolek committed:
fixed tests

petrparolek committed:
fixed code reported by PHPStan

petrparolek committed:
upgrade PHPStan

f3l1x commented on .editorconfig:
```diff
@@ -14,3 +14,9 @@ tab_width = 4
 indent_style = space
 indent_size = 2
+
+[tests/fixtures/files/interactive.json]
```
Proc je tu tohle?

petrparolek replied:
Protože IDE mi neustále přepisoval soubor a padaly testy. Máš jiný nápad na opravu?

f3l1x replied:
Jak ti to prepisovalo testy? A jak padaly?

petrparolek replied:
když vrátím editorconfig do původního souboru, dám v IDE Reformat code, uložím soubor, tak nastane nesmysl v testech  ![obrazek](https://user-images.githubusercontent.com/6066243/148379969-ab59bcbe-1ec8-44df-9ef4-5e0805129028.png)

[output.zip](https://github.com/contributte/neonizer/files/7821551/output.zip)

f3l1x commented on src/TaskSet.php:
```diff
@@ -72,6 +72,8 @@ public function set(array $args): void
 			// Black magic. It parse --database.host=localhost into $tmp array.
 			// Easy to use! :-)
+			/** @var string $value */
+			$value = $value;
```
Stacilo by `parse_str('parameters[' . str_replace('.', '][', $key) . ']=' . (string) $value, $tmp);`

petrparolek replied:
OK. Zkusím.

petrparolek replied:
```
 ------ --------------------------------------------------------- 
  Line   TaskSet.php                                              
 ------ --------------------------------------------------------- 
  75     Cannot cast array|bool|float|int|string|null to string.  
 ------ --------------------------------------------------------- 
```

petrparolek replied:
https://github.com/petrparolek/contributte-neonizer/actions/runs/1662846839

f3l1x replied:
Tak to chce vymyslet neco jineho. `$value = $value;` je blbost.

petrparolek replied:
Nejlép nám poradí @ondrejmirtes

ondrejmirtes replied:
Ahoj, prosim, jestli chcete poradit, zreprodukujte problem na phpstan.org, a idealne se zeptejte anglicky v Discussions :) https://github.com/phpstan/phpstan/discussions Diky.

f3l1x replied:
Nezlob se @ondrejmirtes. :-)

ondrejmirtes replied:
Ja se nezlobim, ale musim nejak managovat svuj cas, navic na anglickou diskuzi muzu pozdeji linkovat kdyz se nekdo zepta na neco podobneho 😊

f3l1x replied:
To ja uplne chapu. Uz jsem domluvil Petrovi, aby te neoznacoval u kravin. :-)

f3l1x commented on .editorconfig:
```diff
@@ -14,3 +14,9 @@ tab_width = 4
 indent_style = space
 indent_size = 2
+
+[tests/fixtures/files/interactive.json]
```
Tohle prosim odmazat

petrparolek replied:
OK

f3l1x commented on Makefile:
```diff
@@ -12,7 +12,7 @@ csf:
 
 phpstan:
-	vendor/bin/phpstan analyse -l max -c phpstan.neon src
+	vendor/bin/phpstan analyse -l 8 -c phpstan.neon src
```
Level a path do neonu

petrparolek replied:
OK

f3l1x commented on src/TaskSet.php:
```diff
@@ -72,6 +72,8 @@ public function set(array $args): void
 		foreach ($input->getOptions() as $key => $value) {
 			// Black magic. It parse --database.host=localhost into $tmp array.
 			// Easy to use! :-)
+			/** @var string $value */
```
Tohle smaz a dej to do ignore.

petrparolek committed:
improvements
- updated composer packages
- added allow-plugins in composer.json
- improved phpstan
